musl strdupa() fails to compile in c++ due to a missing cast
use deadbeef's fallback version instead

--- strdupa.h
+++ strdupa.h
@@ -26,6 +26,10 @@
 
 #include <string.h>
 
+#if defined(__linux__) && !defined(__GLIBC__)
+# undef strdupa
+#endif
+
 #ifndef strdupa
 # define strdupa(s)                                                          \
     ({                                                                       \
